feature/products

pawelrogowski opened the pull request:
added:
-html-markup for product section in index.html
-scss partial for product section
-scss partial for sections
-3dots.png for pseudoelement
-@use for product partial in main.scss

pawelrogowski committed:
added product section

added:
-html-markup for product section in index.html
-scss partial for product section
-scss partial for sections
-3dots.png for pseudoelement
-@use for product partial in main.scss

pkgorniak commented on src/partials/products.html:
```diff
@@ -0,0 +1,44 @@
+<section class="section__products">
```
Potrzebujemy kotwicę, żeby link z menu przenosił do tej sekcji. W tagu section dodaj ` id="products" `.
Kotwica może być tutaj albo na naglówku "100% natural".

pkgorniak commented on src/partials/products.html:
```diff
@@ -0,0 +1,44 @@
+          ut labore et. dolore magna aliqua facilisis.
+        </p>
+        <button class="products__button" type="button">
+          <svg class="products__button-icon" width="40" height="40">
```
width ikony jest zdublowana/nadpisana w css. myślę, że można wyrzucić width i height dla ikon z html i sterować z css.
![obraz](https://user-images.githubusercontent.com/109943174/195776491-355ec6d0-4517-44d6-9e92-cb916e70426a.png)

pkgorniak commented on src/sass/_products.scss:
```diff
@@ -0,0 +1,317 @@
+    );
+    flex-basis: 100%;
+    &--ice-cream {
+      background-color: #ffb8ca;
```
Spróbuj proszę do `&--ice-cream` dodać  `@extend .products__item;` i zobacz czy html będzie działał tylko z jedną klasą -> `products__item--ice-cream`
testowałem u siebie, działa.
ten sam temat z `--coffee` i `--milkshake`

pkgorniak commented on src/sass/_products.scss:
```diff
@@ -0,0 +1,317 @@
+      $weight: 400,
+      $style: normal
+    );
+    font-family: 'Titan One';
```
font-family możesz zaciągać z partiala _variables: `font-family: var.$brand-font-family;`

pkgorniak commented on src/sass/_products.scss:
```diff
@@ -0,0 +1,317 @@
+      $weight: 700,
+      $style: normal
+    );
+    font-family: 'DM Sans';
```
główny font-family jest dodany do body w partialu _global, także nie powinno być potrzeby powtarzania go tutaj

pkgorniak commented on src/sass/_products.scss:
```diff
@@ -0,0 +1,317 @@
+    cursor: pointer;
+    overflow: hidden;
+    transition: background-color 250ms;
+    &:hover {
```
hover i focus jezeli mają takie same właściwości można zgrupować:
`&:hover,
&:focus {
background-color: #ecebeb;
transition: background-color 250ms;
}`

pkgorniak commented on src/sass/_products.scss:
```diff
@@ -0,0 +1,317 @@
+      background-color: #ecebeb;
+      transition: background-color 250ms;
+    }
+    &:hover .products__button-icon {
```
Usuń proszę zmianę filla strzałki na hoverze i focusie żeby zachować zgodność z projektem, tylko zmiana tła na szare.
![obraz](https://user-images.githubusercontent.com/109943174/195778757-8ec6a13f-db71-410d-8d59-7d12134f976c.png)

pawelrogowski committed:
minor fixes and adjustments

Added:
-id anchor for the section
-@extend for list items

Removed:
-duplicated width&height for icons (now only in scss)
-icon fill
-redundant classes in html
-redundant font-family property
-duplicated properties for :hover&:focus

Adjusted:
-replaced font properties with variables

pkgorniak committed:
Merge branch 'main' into feature/products
